products/cloudiot/terraform.yaml

```diff
@@ -0,0 +1,141 @@
+        name: "cloudiot_device_registry_single_event_notification_configs"
+        primary_resource_id: "test-registry"
+        vars:
+          cloudiot_registry_name: "cloudiot-registry"
+          cloudiot_device_telemetry_topic_name: "default-telemetry"
+        test_env_vars:
+          project: :PROJECT_NAME
+          region: :REGION
+      - !ruby/object:Provider::Terraform::Examples
+        name: "cloudiot_device_registry_full"
+        primary_resource_id: "test-registry"
+        vars:
+          cloudiot_registry_name: "cloudiot-registry"
+          cloudiot_device_status_topic_name: "default-devicestatus"
+          cloudiot_device_telemetry_topic_name: "default-telemetry"
+          cloudiot_additional_device_telemetry_topic_name: "additional-telemetr"
+          cloudiot_subfolder_matches_additional_device_telemetry_topic: "test/path"
+        test_env_vars:
+          project: :PROJECT_NAME
+          region: :REGION
+    properties:
+      id: !ruby/object:Overrides::Terraform::PropertyOverride
+        required: true
+        name: 'name'
+        description: |
+          A unique name for the resource, required by device registry.
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validateCloudIotDeviceRegistryId'
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        # We don't need this field, because it has the same format as the ID
+        exclude: true
+      region: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+        required: false
+        default_from_api: true
+        description: |
+          The region in which the created registry should reside.
+          If it is not provided, the provider region is used.
+      logLevel: !ruby/object:Overrides::Terraform::PropertyOverride
+        name: 'log_level'
+        diff_suppress_func: 'emptyOrDefaultStringSuppress("")'
+      eventNotificationConfigs: !ruby/object:Overrides::Terraform::PropertyOverride
+        name: 'event_notification_configs'
+        default_from_api: true
+      eventNotificationConfigs.subfolderMatches: !ruby/object:Overrides::Terraform::PropertyOverride
+        name: 'subfolder_matches'
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validateCloudIotDeviceRegistrySubfolderMatch'
+      eventNotificationConfigs.pubsubTopicName: !ruby/object:Overrides::Terraform::PropertyOverride
+        name: 'pubsub_topic_name'
+        diff_suppress_func: 'compareSelfLinkOrResourceName'
+      stateNotificationConfig: !ruby/object:Overrides::Terraform::PropertyOverride
+        # Excluding this because the original (manually-generated) implementation
+        # wrongly set this to be a Map, instead of a NestedObject. To avoid breaking
+        # changes, we observe that behaviour by excluding this field and adding a
+        # corresponding custom element (of type Map) to the schema. When we're
```

Comment:
It's not just a custom schema entry that you have to add, but also the Create, Read, and Update logic for that entry in encoders/decoders. If you take a look at the generated code, you can see that it doesn't actually include any logic around the new fields; it just puts them in the schema.

Just as a reminder, in case the amount of work is getting to be too much: you don't have to do this. It's totally fine to leave the resource as handwritten and add new features to it there; we do this all the time.

Reply:
Right! Ok, will have a look at this. Do you know if there are other resources that are doing a similar thing, that I can use as an example? Thanks!

Reply:
You can take a look at the encoders/ and decoders/ folder to see, but I'm not sure any of them will be as extensive as you'd need for this. Normally we would just leave this resource handwritten.

Reply:
done